nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/hesai_decoder.hpp

```diff
@@ -36,6 +38,85 @@
+  uint64_t distance_filtered_count = 0;
+  uint64_t fov_filtered_count = 0;
+  uint64_t invalid_point_count = 0;
+  uint64_t identical_filtered_count = 0;
+  uint64_t multiple_return_filtered_count = 0;
+  uint64_t total_kept_point_count = 0;
+  uint64_t invalid_packet_count = 0;
+  float cloud_distance_min_m = std::numeric_limits<float>::infinity();
+  float cloud_distance_max_m = std::numeric_limits<float>::lowest();
+  float cloud_azimuth_min_deg = std::numeric_limits<float>::infinity();
+  float cloud_azimuth_max_rad = std::numeric_limits<float>::lowest();
+  uint64_t packet_timestamp_min_ns = std::numeric_limits<uint64_t>::max();
+  uint64_t packet_timestamp_max_ns = std::numeric_limits<uint64_t>::min();
+
+  [[nodiscard]] nlohmann::ordered_json to_json() const
+  {
+    nlohmann::json distance_j;
+    distance_j["name"] = "distance";
+    distance_j["filtered_count"] = distance_filtered_count;
+    // distance_j["cloud_distance_min_m"] = cloud_distance_min_m;
+    // distance_j["cloud_distance_max_m"] = cloud_distance_max_m;
+    nlohmann::json fov_j;
+    fov_j["name"] = "fov";
+    fov_j["filtered_count"] = fov_filtered_count;
+    // fov_j["cloud_azimuth_min_deg"] = cloud_azimuth_min_deg;
+    // fov_j["cloud_azimuth_max_rad"] = cloud_azimuth_max_rad;
+    nlohmann::json identical_j;
+    identical_j["name"] = "identical";
+    identical_j["filtered_count"] = identical_filtered_count;
+    nlohmann::json multiple_j;
+    multiple_j["name"] = "multiple";
+    multiple_j["filtered_count"] = multiple_return_filtered_count;
+    nlohmann::json invalid_j;
+    invalid_j["name"] = "invalid";
+    invalid_j["invalid_point_count"] = invalid_point_count;
+    invalid_j["invalid_packet_count"] = invalid_packet_count;
+    nlohmann::json pointcloud_bounds_azimuth_j;
+    pointcloud_bounds_azimuth_j["min"] = cloud_azimuth_min_deg;
+    pointcloud_bounds_azimuth_j["max"] = cloud_azimuth_max_rad;
+    nlohmann::json pointcloud_bounds_distance_j;
+    pointcloud_bounds_distance_j["min"] = cloud_distance_min_m;
+    pointcloud_bounds_distance_j["max"] = cloud_distance_max_m;
+    nlohmann::json pointcloud_bounds_timestamp_j;
+    pointcloud_bounds_timestamp_j["min"] = packet_timestamp_min_ns;
+    pointcloud_bounds_timestamp_j["max"] = packet_timestamp_max_ns;
+
+    nlohmann::json j;
+    j["filter_pipeline"] = nlohmann::json::array({
+      distance_j,
+      fov_j,
+      identical_j,
+      multiple_j,
+    });
+    j["pointcloud_bounds"] = {
+      j["azimuth_deg"] = pointcloud_bounds_azimuth_j,
+      j["distance_m"] = pointcloud_bounds_distance_j,
+      j["timestamp_ns"] = pointcloud_bounds_timestamp_j,
+    };
+    j["invalid_filter"] = invalid_j;
+    j["total_kept_point_count"] = total_kept_point_count;
+
+    return j;
+  }
+
+  void update_pointcloud_bounds(const NebulaPoint & point)
+  {
+    cloud_azimuth_min_deg = std::min(cloud_azimuth_min_deg, point.azimuth);
+    cloud_azimuth_max_rad = std::max(cloud_azimuth_max_rad, point.azimuth);
```

Comment:
Please implement [this comment](https://github.com/tier4/nebula/pull/247#discussion_r1903700544).